Sanity/omprog/signal_HUP/runtest.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Dalibor Pospisil <dapospis@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Copyright (c) 2024 Red Hat, Inc.
+#
+#   This copyrighted material is made available to anyone wishing
+#   to use, modify, copy, or redistribute it subject to the terms
+#   and conditions of the GNU General Public License version 2.
+#
+#   This program is distributed in the hope that it will be
+#   useful, but WITHOUT ANY WARRANTY; without even the implied
+#   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#   PURPOSE. See the GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public
+#   License along with this program; if not, write to the Free
+#   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+#   Boston, MA 02110-1301, USA.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+PACKAGE="rsyslog"
+
+rlJournalStart && {
+  rlPhaseStartSetup && {
+    rlRun "rlImport --all" 0 "Import libraries" || rlDie "cannot continue"
+    rlRun "rlCheckMakefileRequires" || rlDie "cannot continue"
+    rlRun "TmpDir=\$(mktemp -d)" 0 "Creating tmp directory"
+    CleanupRegister "rlRun 'rm -r $TmpDir' 0 'Removing tmp directory'"
+    CleanupRegister 'rlRun "popd"'
+    rlRun "pushd $TmpDir"
+    CleanupRegister 'rlRun "rsyslogCleanup"'
+    rlRun "rsyslogSetup"
+    CleanupRegister 'rlRun "rlSEBooleanRestore domain_can_mmap_files logging_syslogd_list_non_security_dirs"'
+    rlRun 'rlSEBooleanOn domain_can_mmap_files logging_syslogd_list_non_security_dirs'
+    CleanupRegister 'rlRun "rlServiceRestore rsyslog"'
+    rlRun "rlServiceStop rsyslog"
+    CleanupRegister 'rlRun "rlFileRestore"'
+    rlRun "rlFileBackup --clean /etc/rsyslog.conf /etc/rsyslog.d/ /usr/libexec/rsyslog/log_rotate.sh /etc/logrotate.d/test /var/log/logs/"
+    rlRun "mkdir -p /usr/libexec/rsyslog"
+    rsyslogConfigAddTo --begin "MODULES" < <(rsyslogConfigCreateSection "MODLOAD OMPROG")
+    rsyslogConfigAddTo "MODLOAD OMPROG" <<EOF
+module(load="omprog")
+EOF
+rsyslogConfigAddTo --begin "RULES" <<EOF
+if (\$programname startswith 'module_execution_') then {
+  action(type="omprog"
+       binary="/usr/libexec/rsyslog/log_rotate.sh")
+}
+EOF
+    rsyslogConfigAddTo "MODLOAD IMUXSOCK" <<EOF
+input(type="imuxsock" Socket="/dev/log2")
+EOF
+    cat > /usr/libexec/rsyslog/log_rotate.sh << 'EOF'
+#!/bin/bash
+exec >> /var/log/logs/test.out
+exec 2>&1
+file=/var/log/logs/test.log
+echo "before SIGHUP" >> "$file"
+#/usr/bin/systemctl --signal=HUP kill rsyslog 2> /dev/null
```

Comment:
```suggestion
#/usr/bin/systemctl --signal=HUP --kill-whom=main kill rsyslog 2> /dev/null
```

Reply:
By default, if a signal is sent to the main process, rsyslog does not forward them to child processes unless it's configured to. However, in case of ```systemctl kill```, it does not send a signal to a single process, rather to all processes started by a service unit. This can be altered by ```--kill-whom=main``` option.